File: Daily Challanges/2026-06-June/2026-06-21.js
function getDaytimeHours(latitude) {
  const daylight = Math.round((12 + (latitude / 90) * 12) / 2) * 2;
  const sunrise = 12 - daylight / 2;
  const sunset = 12 + daylight / 2;

  let result = "";

  for (let h = 0; h < 24; h++) {
    result += h >= sunrise && h < sunset ? "☀️" : "🌑";
  }

  return result;
}

console.log(getDaytimeHours(0));
console.log(getDaytimeHours(90));
console.log(getDaytimeHours(-90));
console.log(getDaytimeHours(-33));
console.log(getDaytimeHours(66.5));
console.log(getDaytimeHours(40));
console.log(getDaytimeHours(-50));

File: Daily Challanges/2026-06-June/2026-06-21.py
def get_daytime_hours(latitude):
    daylight = round((12 + (latitude / 90) * 12) / 2) * 2
    sunrise = 12 - daylight / 2
    sunset = 12 + daylight / 2

    result = ""

    for h in range(24):
        result += "☀️" if sunrise <= h < sunset else "🌑"

    return result


print(get_daytime_hours(0))
print(get_daytime_hours(90))
print(get_daytime_hours(-90))
print(get_daytime_hours(-33))
print(get_daytime_hours(66.5))
print(get_daytime_hours(40))
print(get_daytime_hours(-50))
